src/core/p3m.cpp

```diff
@@ -361,7 +361,7 @@ void p3m_init() {
 
     /* position offset for calc. of first meshpoint */
     p3m.pos_shift =
-        (double)((p3m.params.cao - 1) / 2) - (p3m.params.cao % 2) / 2.0;
+        std::floor((p3m.params.cao - 1) / 2) - (p3m.params.cao % 2) / 2.0;
```

Comment:
Not sure if this change (and the same one in p3m-dipolar.cpp) matches what was intended. I don't understand the surrounding code. It silences the warning while preserving old behavior, but looks odd.

Reply:
The floor does not do anything, as this is integer division. In p3m-dipolar you made the `2` a float. If this works it's not clear to me what raised the warning in the first place. (I *think* btw that the integer division is intended and the old code was correct).

Reply:
The warning was about casting an integer dicision to float. There's nothing wrong per se, but most of the time such an operation is not intended. The floor keeps it consistently as float, but you are right, I forgot to make the 2 a 2.0 here.